paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/PaymentMethodListService.java

```diff
@@ -33,4 +40,62 @@
 			return new PaymentMethodDto[0];
 		}
 	}
+
+	public PaymentMethodDto[] filterPaymentMethodList(GetPaymentMethodListRequest request, PaymentMethodDto[] methods) {
+		String namespace = request.getNamespace();
+		List<PaymentMethodFilter> filtersForNamespace = PaymentMethodFilter.getByNamespace(namespace);
+
+		Set<PaymentMethodDto> filteredMethodsList = new HashSet<>();
+
+		for (PaymentMethodFilter paymentMethodFilter : filtersForNamespace) {
+			String filterKey = getFilterKeys(paymentMethodFilter, request);
+
+			HashMap<String, List<String>> filterValues = getFilterValuesMap(paymentMethodFilter);
+
+			for (Map.Entry<String, List<String>> entry : filterValues.entrySet()) {
+				String key = entry.getKey();
+
+				if (filterKey.equalsIgnoreCase(key)) {
+					List<String> paymentMethodGroups = entry.getValue();
+
+					for (String value : paymentMethodGroups) {
+						filteredMethodsList.addAll(Arrays.stream(methods)
+								.filter(method -> method.getGroup().equalsIgnoreCase(value))
+								.collect(Collectors.toList()));
+					}
+				}
+			}
+		}
+
+		if (!filteredMethodsList.isEmpty()) {
+			methods = filteredMethodsList.toArray(new PaymentMethodDto[0]);
+		}
+
+		return methods;
+	}
+
+	public String getFilterKeys(PaymentMethodFilter filter, GetPaymentMethodListRequest request) {
+		OrderDto orderDto = request.getOrderDto();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			return orderDto.getType();
+		}
+
+		throw new CommonApiException(
+				HttpStatus.INTERNAL_SERVER_ERROR,
+				new Error("unknown-payment-method-filter-type", "unknown payment method filter type")
+		);
+	}
+
+	// namespace can be used as a parameter if different namespaces need to return different values for filter type
+	public HashMap<String, List<String>> getFilterValuesMap(PaymentMethodFilter filter) {
+		HashMap<String, List<String>> values = new HashMap<>();
+
+		if (filter.equals(PaymentMethodFilter.ORDER_TYPE)) {
+			values.put("subscription", Collections.singletonList("creditcards"));
```

Comment:
The available keywords that can be used as filters (and are specific to Visma response) should be listed as constants. We should probably have a keywords/constants section in common package for order type.

Reply:
You can make one for PaymentType.CREDIT_CARDS = "creditcards"